code/game/machinery/doors/airlock.dm

```diff
@@ -1604,6 +1605,30 @@
 	else if(istype(note, /obj/item/photo))
 		return "photo"
 
+/obj/machinery/door/airlock/AltClick(mob/living/user)
+	if(!istype(user))
+		to_chat(user, "<span class='info'>Nice try, ghosts.</span>")
+		return
+
+	if (!user.canUseTopic(src))
+		to_chat(user, "<span class='info'>You can't do this right now!</span>")
+		return
+
+	if(stat & (NOPOWER|BROKEN) || emagged)
+		to_chat(user, "<span class='info'>The door isn't working!</span>")
+		return
+
+	if(request_cooldown > world.time)
+		to_chat(user, "<span class='info'>The airlocks spam filter is blocking your request. Please wait at least 10 seconds between requests.</span>")
+		return
+
+	for(var/mob/living/silicon/ai/AI in GLOB.living_mob_list)
+		if(!AI.client)
+			continue
+		to_chat(AI, "<span class='info'>[user.name] is requesting you to open [src]<a href='?src=\ref[AI];remotedoor=\ref[src]'>(Open)</a></span>")
```

Comment:
Needs a jump link

Reply:
it should also state the job of whoever is requesting, and the name should be a link to their criminal record.

Reply:
Hijacking your comment too. Even though I ___assume___ the AI cannot open a door via the request if the AI wire is cut, should the request be sent through in that case?